### PR TITLE
Add test manifest for OpenSearch Dashboards 1.3.0

### PR DESCRIPTION
**Pull request by tianleh**
Signed-off-by: Tianle Huang <tianleh@amazon.com>

### Description
This test manifest was previously removed by https://github.com/opensearch-project/opensearch-build/pull/1470 when OSD 1.3.0 was to be replaced by OSD 2.0.0

Looks like now we are releasing OSD 1.3.0 https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.3.0/opensearch-dashboards-1.3.0.yml

Thus add back this test manifest file. Dashboards plugin teams are actively onboarding on to the FTRepo https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/72
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaiksaya — APPROVED** [review]

**Review by zelinh — APPROVED** [review]

File: manifests/1.3.0/opensearch-dashboards-1.3.0-test.yml
Comment (kavilla) [inline manifests/1.3.0/opensearch-dashboards-1.3.0-test.yml:9]: I think we like putting the schema-version on line 2.

**Review by kavilla — APPROVED** [review]

**Comment by tianleh** [comment]
See this check failure.
https://github.com/opensearch-project/opensearch-build/runs/5279350076?check_suite_focus=true

This is related to `  ./ci.sh manifests/1.3.0/opensearch-1.3.0.yml --snapshot` which is irrelevant of this PR.

**Comment by tianleh** [comment]
Created a bug https://github.com/opensearch-project/opensearch-build/issues/1661

File: manifests/1.3.0/opensearch-dashboards-1.3.0-test.yml
Comment (tianleh) [reply manifests/1.3.0/opensearch-dashboards-1.3.0-test.yml]: updated

**Comment by tianleh** [comment]
After merging the latest main branch, all checks are successful. Could you @gaiksaya @zelinh please help click the merge button?

**Pull request merged**
